packages/pd-manager-rest-api/src/api-functions.ts

```diff
@@ -0,0 +1,94 @@
+import { checkAuth, sendErrorResponse, ISingleEndpointOpts } from '@sphereon/ssi-express-support'
+import { Request, Response, Router } from 'express'
+import { IRequiredContext } from './types'
+import { AddDefinitionItemArgs, PersistDefinitionArgs, UpdateDefinitionItemArgs } from '@sphereon/ssi-sdk.pd-manager'
+import { DeleteDefinitionArgs } from '@sphereon/ssi-sdk.data-store'
+
+export function pdReadEndpoint(router: Router, context: IRequiredContext, opts?: ISingleEndpointOpts) {
+  if (opts?.enabled === false) {
+    console.log(`"pdReadEndpoint" Endpoint is disabled`)
+    return
+  }
+  const path = opts?.path ?? '/presentation-defs'
+  router.get(`${path}/:pdId`, checkAuth(opts?.endpoint), async (request: Request, response: Response) => {
+    try {
+      const pdId = request.params.pdId
+      const pd = await context.agent.pdmGetDefinition({ itemId: pdId })
+      response.statusCode = 200
+      return response.send(pd)
+    } catch (error) {
+      return sendErrorResponse(response, 500, error.message as string, error)
+    }
+  })
+}
+
+export function pdAddEndpoint(router: Router, context: IRequiredContext, opts?: ISingleEndpointOpts) {
+  if (opts?.enabled === false) {
+    console.log(`"pdAddEndpoint" Endpoint is disabled`)
+    return
+  }
+  const path = opts?.path ?? '/presentation-defs'
+  router.post(path, async (request: Request, response: Response) => {
+    try {
+      const addPd = request.body
+      const pd = await context.agent.pdmAddDefinition(addPd as AddDefinitionItemArgs)
+      response.statusCode = 201
+      return response.send(pd)
+    } catch (error) {
+      return sendErrorResponse(response, 500, error.message, error)
+    }
+  })
+}
+
+export function pdUpdateEndpoint(router: Router, context: IRequiredContext, opts?: ISingleEndpointOpts) {
```

Comment:
should this one not be on id? update a certain resource?

Reply:
Fixed + check